dnf/base.py

```diff
@@ -2298,14 +2299,33 @@ def _prov_key_data(msg):
                     logger.info(msg, keyurl, info.short_id)
                     continue
 
+                # DNS Extension: create a key object, pass it to the verification class
+                # and print its result as an advice to the user.
+                if self.conf.dns_verification:
+                    dns_input_key = dnssec.KeyInfo.from_rpm_key_object(info.userid, info.raw_key)
+                    dns_result = dnssec.DNSSECKeyVerification.verify(dns_input_key)
+                    logger.info(dnssec.nice_user_msg(dns_input_key, dns_result))
+
                 # Try installing/updating GPG key
                 info.url = keyurl
                 dnf.crypto.log_key_import(info)
                 rc = False
                 if self.conf.assumeno:
                     rc = False
                 elif self.conf.assumeyes:
-                    rc = True
+                    # DNS Extension: We assume, that the key is trusted in case it is valid,
+                    # its existence is explicitly denied or in case the domain is not signed
+                    # and therefore there is no way to know for sure (this is mainly for
+                    # backward compatibility)
+                    if self.conf.dns_verification:
+                        if dns_result in (dnssec.Validity.VALID, dnssec.Validity.PROVEN_NONEXISTENCE):
```

Comment:
add a comment what PROVEN_NONEXISTENCE means
it doesn't make sense for someone who doesn't understand dnssec